src/test/java/io/kestra/plugin/microsoft365/sharepoint/UploadTest.java

```diff
@@ -0,0 +1,219 @@
+package io.kestra.plugin.microsoft365.sharepoint;
+
+import com.microsoft.graph.drives.item.items.item.DriveItemItemRequestBuilder;
+import com.microsoft.graph.drives.item.DriveItemRequestBuilder;
+import com.microsoft.graph.drives.DrivesRequestBuilder;
+import com.microsoft.graph.models.DriveItem;
+import com.microsoft.graph.serviceclient.GraphServiceClient;
+import io.kestra.core.junit.annotations.KestraTest;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.runners.RunContextFactory;
+import io.kestra.core.storages.StorageInterface;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.net.URI;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.doReturn;
+
+@KestraTest
+class UploadTest {
+    @Inject
+    private RunContextFactory runContextFactory;
+
+    @Inject
+    private StorageInterface storageInterface;
+
+    private RunContext runContext;
```

Comment:
It's not a good idea to have a shared `runContext` between all the tests, since it can lead to flaky tests. Could you remove it and have a test scoped runContext through the factory please? And this for every single test 🙏 Thanks!